src/prisma/schema.prisma

```diff
@@ -1939,3 +1941,49 @@ model AdditionalBlockQuota {
   blockCount Int
   createdAt  DateTime  @default(now())
 }
+
+model Form {
+  id         String      @id @default(uuid()) @db.Uuid
+  formFields FormField[]
+  proposal   Proposal[]
+}
+
+enum FormFieldType {
+  short_text
+  long_text
+  number
+  email
+  phone
+  url
+  select
+  multiselect
+  wallet
+  date
+  person
+  label
+}
+
+model FormField {
+  id            String           @id @default(uuid()) @db.Uuid
+  type          FormFieldType
+  index         Int              @default(-1)
+  name          String
+  description   String?
+  required      Boolean          @default(false)
+  private       Boolean          @default(false)
+  options       Json?
+  parentFieldId String?          @db.Uuid
+  formId        String           @db.Uuid
+  form          Form             @relation(fields: [formId], references: [id], onDelete: Cascade)
+  fieldValues   FormFieldValue[]
+}
+
+model FormFieldValue {
+  id      String        @id @default(uuid()) @db.Uuid
+  fieldId String        @db.Uuid
+  field   FormField     @relation(fields: [fieldId], references: [id], onDelete: Cascade)
+  type    FormFieldType
+  value   Json
+
```

Comment:
I think we should have here also relation to proposal:

```
  proposalId       String                 @db.Uuid
  proposal         Proposal              @relation(fields: [proposalId], references: [id], onDelete: Cascade)
  ```

and in proposal:
```
formAsnwers: ProposalFormAnswers[]
```

this way it will be much easier to actually query for answers of a given proposal

Reply:
Agree, we need to have this relation, otherwise which answers belong to a proposal won't be apparent